### DISCUSSION TITLE
How to achieve nested style changes?

### DISCUSSION THREAD

**Discussion by xD0135** [AvaloniaUI/Avalonia#10519] [Q&A]
What I'm trying to do is change the style of a few elements from a `ListBoxItem` when an item is selected, but I'm not sure how to do it or if it's even possible. 

My `ListBox` is defined as such:
```
               <ListBox
                    Grid.Row="1"
                    Items="{Binding SidebarMenuPrimary}"
                    SelectionMode="Single">
                    <ListBox.ItemTemplate>
                        <DataTemplate>
                            <StackPanel>
                                <PathIcon Data="{Binding Icon}" />
                                <TextBlock Text="{Binding Text}" />
                            </StackPanel>
                        </DataTemplate>
                    </ListBox.ItemTemplate>
                </ListBox>
```

And what I would like to do is set something like this (which doesn't work), gives me compiler error:
```
            <Style Selector="ListBoxItem:selected /template/ ContentPresenter">
                <Setter Property="Background" Value="Green" />
                 <-- This part fails, regardless of what I use as the selector -->
                <Style Selector="ListBoxItem > StackPanel > TextBlock">
                    <Setter Property="FontSize" Value="16" />
                </Style>
            </Style>
```
I found https://github.com/AvaloniaUI/Avalonia/issues/3593 and thought it would work but it doesn't. 

I'm stuck and any help is appreciated. Thanks

**Comment by xD0135** [comment]
Can anyone help answer my question? Thanks.

**Comment by timunie** [comment]
Please let me know the version of Avalonia you use. 

Note: I really think the Style should not be nested inside another Style. Probably this was already the issue?

```xml
  <Style Selector="ListBoxItem:selected /template/ ContentPresenter">
      <Setter Property="Background" Value="Green" />
       <!-- /!\ This should not be placed here. -->
  </Style>

  <!-- This Style goes here. Use DevTools [F12] to check the visual Tree -->
  <Style Selector="ListBoxItem > StackPanel > TextBlock">
      <Setter Property="FontSize" Value="16" />
  </Style>
```

This is the output I have: 

![image](https://user-images.githubusercontent.com/47110241/223049352-f17f59bd-2a5d-4e35-9d54-4a6e87011f3a.png)

**Reply by xD0135** [reply]
Thank you for your answer. The version I have is `0.10.18`.

What I'm trying to do is different than what you have presented, which is to cascade the style changes across multiple elements upon the selection of the row. For example, when the selector is `Selector="ListBoxItem:selected /template/ ContentPresenter"`, set the `FontSize` or `FontWeight` of the selected row's `TextBlock` to a different value then the pre-defined style. If I think from a CSS perspective, this would be similar to applying a specific `.class` to an element upon an action. Is this something that can be done?

**Reply by timunie** [reply]
In limited scope, yes. Best is to explore the existing Styles. Here is the FluentTheme for version 0.10.18: https://github.com/AvaloniaUI/Avalonia/tree/release/0.10.18/src/Avalonia.Themes.Fluent/Controls

**Reply by xD0135** [reply]
Thank you. Reading through the existing styles I don't understand what needs to be done in order to achieve the above. I tried the following but the results were still the same, meaning that it didn't change anything:

```
<Style Selector="ListBoxItem:selected /template/ ContentPresenter">
  <Setter Property="Background" Value="white" />
  <Setter Property="TextBlock.FontWeight" Value="UltraBlack" />
</Style>
```
Are there other resources or documentation that you can point me to?

**Reply by timunie** [reply]
https://docs.avaloniaui.net/docs/styling/selectors 

And 

https://docs.avaloniaui.net/docs/getting-started/developer-tools
